src/AddToMetaMask.tsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+
+const AddToMetaMask = (props: {
+  name: string,
+  chainId: string,
+  token: string,
+  rpcs: string[],
+  be: string[],
+}) => {
+  return (
+    <span>
+      {props.rpcs.map(item => (
+        <code key={item}>{item}<br/></code>
+      ))}
+      <button
+        className="button button--primary margin-top--md"
+        onClick={() => {
+          if (!window.ethereum?.request) {
+            return alert('Have you installed MetaMask yet? If not, please do so.\n\nComputer: Once it is installed, you will be able to add the ParaTime to your MetaMask.\n\nPhone: Open the website through your MetaMask Browser to add the ParaTime.')
+          }
+          window.ethereum.request({
+            method: 'wallet_addEthereumChain',
+            params: [
+              {
+                chainId: props.chainId,
+                chainName: props.name,
+                nativeCurrency: {
+                  name: props.token,
+                  symbol: props.token,
+                  decimals: 18,
+                },
+                rpcUrls: props.rpcs,
+                blockExplorerUrls: props.be,
+              },
+            ],
+          })
+        }}
+      >
+        Add to MetaMask
+      </button>
+    </span>
+  );
+};
+
+export const AddSapphireToMetaMask = (props: {
+  rpcs: string[],
+}) => {
+  return AddToMetaMask({
+    name: 'Oasis Sapphire',
+    chainId: '0x5afe',
+    token: 'ROSE',
+    rpcs: props.rpcs,
+    be: ['https://explorer.sapphire.oasis.io'],
```

Comment:
NOTE: currently blockscout is available on this URL, but it will be migrated to: https://old-explorer.sapphire.oasis.io/ (the new URLalready works). The explorer.sapphire.oasis.io will start pointing to Nexus.

We should switch this to either old-explorer.sapphire.oasis.io or maybe use nexus here already? 

BTW did anyone check, does nexus work with metamask generated links?

Reply:
> BTW did anyone check, does nexus work with metamask generated links?

Yes, such explorer URL scheme is defined by [EIP-3091](https://eips.ethereum.org/EIPS/eip-3091) and we are compatible with it.